$NetBSD: patch-ocamldoc_Makefile,v 1.2 2014/10/09 19:08:28 jaapb Exp $

Use BSD_INSTALL_* instead of mkdir and cp; install in man3o
--- ocamldoc/Makefile.orig	2014-07-21 15:44:24.000000000 +0000
+++ ocamldoc/Makefile
@@ -41,7 +41,7 @@ INSTALL_LIBDIR=$(DESTDIR)$(OCAMLLIB)/oca
 INSTALL_CUSTOMDIR=$(INSTALL_LIBDIR)/custom
 INSTALL_BINDIR=$(DESTDIR)$(OCAMLBIN)
 #MANO: man ocamldoc
-INSTALL_MANODIR=$(DESTDIR)$(MANDIR)/man3
+INSTALL_MANODIR=$(DESTDIR)$(MANDIR)/man3o
 
 INSTALL_MLIS=odoc_info.mli
 INSTALL_CMIS=$(INSTALL_MLIS:.mli=.cmi)
@@ -240,24 +240,24 @@ odoc_see_lexer.ml: odoc_see_lexer.mll
 # Installation targets
 ######################
 install: dummy
-	if test -d $(INSTALL_BINDIR); then : ; else $(MKDIR) $(INSTALL_BINDIR); fi
-	if test -d $(INSTALL_LIBDIR); then : ; else $(MKDIR) $(INSTALL_LIBDIR); fi
-	if test -d $(INSTALL_CUSTOMDIR); then : ; else $(MKDIR) $(INSTALL_CUSTOMDIR); fi
-	$(CP) $(OCAMLDOC) $(INSTALL_BINDIR)/$(OCAMLDOC)$(EXE)
-	$(CP) ocamldoc.hva *.cmi $(OCAMLDOC_LIBCMA) $(INSTALL_LIBDIR)
-	$(CP) $(INSTALL_MLIS) $(INSTALL_CMIS) $(INSTALL_LIBDIR)
-	if test -d $(INSTALL_MANODIR); then : ; else $(MKDIR) $(INSTALL_MANODIR); fi
-	if test -d stdlib_man; then $(CP) stdlib_man/* $(INSTALL_MANODIR); else : ; fi
+	if test -d $(INSTALL_BINDIR); then : ; else $(BSD_INSTALL_PROGRAM_DIR) $(INSTALL_BINDIR); fi
+	if test -d $(INSTALL_LIBDIR); then : ; else $(BSD_INSTALL_DATA_DIR) $(INSTALL_LIBDIR); fi
+	if test -d $(INSTALL_CUSTOMDIR); then : ; else $(BSD_INSTALL_DATA_DIR) $(INSTALL_CUSTOMDIR); fi
+	$(BSD_INSTALL_SCRIPT) $(OCAMLDOC) $(INSTALL_BINDIR)/$(OCAMLDOC)$(EXE)
+	$(BSD_INSTALL_DATA) ocamldoc.hva *.cmi $(OCAMLDOC_LIBCMA) $(INSTALL_LIBDIR)
+	$(BSD_INSTALL_DATA) $(INSTALL_MLIS) $(INSTALL_CMIS) $(INSTALL_LIBDIR)
+	if test -d $(INSTALL_MANODIR); then : ; else $(BSD_INSTALL_MAN_DIR) $(INSTALL_MANODIR); fi
+	if test -d stdlib_man; then $(BSD_INSTALL_MAN) stdlib_man/* $(INSTALL_MANODIR); else : ; fi
 
 installopt:
 	if test -f $(OCAMLDOC_OPT); then $(MAKE) installopt_really ; fi
 
 installopt_really:
-	if test -d $(INSTALL_BINDIR); then : ; else $(MKDIR) $(INSTALL_BINDIR); fi
-	if test -d $(INSTALL_LIBDIR); then : ; else $(MKDIR) $(INSTALL_LIBDIR); fi
-	$(CP) $(OCAMLDOC_OPT) $(INSTALL_BINDIR)/$(OCAMLDOC_OPT)$(EXE)
-	$(CP) ocamldoc.hva $(OCAMLDOC_LIBA) $(OCAMLDOC_LIBCMXA) $(INSTALL_LIBDIR)
-	$(CP) $(INSTALL_MLIS) $(INSTALL_CMIS) $(INSTALL_LIBDIR)
+	if test -d $(INSTALL_BINDIR); then : ; else $(BSD_INSTALL_PROGRAM_DIR) $(INSTALL_BINDIR); fi
+	if test -d $(INSTALL_LIBDIR); then : ; else $(BSD_INSTALL_DATA_DIR) $(INSTALL_LIBDIR); fi
+	$(BSD_INSTALL_PROGRAM) $(OCAMLDOC_OPT) $(INSTALL_BINDIR)/$(OCAMLDOC_OPT)$(EXE)
+	$(BSD_INSTALL_DATA) ocamldoc.hva $(OCAMLDOC_LIBA) $(OCAMLDOC_LIBCMXA) $(INSTALL_LIBDIR)
+	$(BSD_INSTALL_DATA) $(INSTALL_MLIS) $(INSTALL_CMIS) $(INSTALL_LIBDIR)
 
 # Testing :
 ###########
